### PR TITLE
Add homepage component

### PR DESCRIPTION
**Pull request by imranhossainemi**
### Description

- [ ] Add Homepage component with demo product list to project from dev branch.

#### General

- [ ] The code follows the project's coding guidelines and best practices.
- [ ] The code is modular, well-organized, and easy to understand.
- [ ] Variable and function names are meaningful and self-descriptive.
- [ ] There are  commented-out code blocks or unused variables/methods for api declaration for future use

#### Functionality

- [ ] The changes accomplish the intended purpose and work as expected.
- [ ] The existing functionality is not affected adversely by the changes.
- [ ] All edge cases and possible error scenarios have been considered.

#### Testing

- [ ] The changes include appropriate unit tests for new functionality have not been tested.
- [ ] Test for responsive design

### CODE REVIEW

File: src/pages/HomePage.jsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+  return (
+    <div className='pro-list'>
+      <h1>Product List</h1>
+      <ul className='products'>
+        {products.map((product) => (
+          <li key={product.id} className='product'>
+            <img src={product.picture} alt={product.name}/>
+            <h2>{product.name}</h2>
+            <button>View Details</button>
```
Comment (shasherazi) [inline src/pages/HomePage.jsx:35]: I think this can be made into its own component for better readability. What do you think?

File: src/index.css
Comment (shasherazi) [inline src/index.css:1]: In my PR for signup/login I actually deleted the `index.css` and preferred `App.css` as all the main stuff was happening there. Let me know what you think. Should we move this code to `App.css` (because it's fairly small code here), or should we keep both?

To avoid unnecessary merge conflicts, you can try copying the CSS in my PR to `App.css`.

**Review by shasherazi — CHANGES_REQUESTED** [review]
Hello @imranhossainemi, good work. However, there is little room for improvement. This is the design we have to follow.
image

And microverse is pretty strict about it.
image

But your design looks like this.
image

I think this design can be made more similar to the given template design.

Also, I commented on minor changes, please see them too.

**Comment by shasherazi** [comment]
Hello @imranhossainemi , sorry to cause you inconvenience. I added the boilerplate/template code in my PR that I pushed just now, causing merge conflicts. To resolve the conflicts in your vscode you can do this.
```
git checkout add_homepage_component
git merge dev
```

And go through the vscode conflict resolver for the conflicted files. Try to stick with the `dev` branch code if possible and merge your code on top of that. You might need to make a few changes to follow the practices, such as using React CSS modules instead of simple CSS to ensure that there are no class name collisions, etc.

File: src/index.css
Comment (imranhossainemi) [reply src/index.css:1]: It need to be there

File: src/pages/HomePage.jsx (hunk 1 above)
Comment (imranhossainemi) [reply src/pages/HomePage.jsx:35]: homepage will empty

**Review by shasherazi — APPROVED** [review]
Hello @imranhossainemi, good work implementing the changes. The PR looks ready to merge :shipit:

**Review by byiringiroscar — APPROVED** [review]
Hi @imranhossainemi 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


* Well written PR 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
